### ISSUE TITLE
List options under 'Type filter' button are not accessible by keyboard: A11y_.NetAspireDashboard_Resource_Filter_Keyboard.

### ISSUE THREAD

**Issue by PoojaNamde** [dotnet/aspire#3173]
"[Check out Accessibility Insights](https://nam06.safelinks.protection.outlook.com/?url=https://accessibilityinsights.io/&data=02%7c01%7cv-stfe%40microsoft.com%7cb67b2c4b646d4f9561a208d6f4b5c39b%7c72f988bf86f141af91ab2d7cd011db47%7c1%7c0%7c636965458850501301&sdata=mxhokIKNMb22llsjXHLgU3XZibj1Qfx37rpY4PU2sfE%3D&reserved=0)! - Identify accessibility bugs before check-in and make bug fixing faster and easier."

**Please do not close this bug. This bug should only be closed by Trusted Tester after verification.**

### Tags:
#A11yTCS; #A11yMAS; #A11ySev2; #Win11; #ChromiumEdge; #FTP; #DesktopWeb; #.NET Core; #SH_.Net Aspire Dashboard_Web_Mar2024; #WCAG2.1.1; #Keyboard;

### Environment Details:
Application Name: .Net Aspire Dashboard
URL: https://localhost:19888/
Chrome Version 122.0.2365.80 (Official build) (64-bit)
Window Version: Win 11
OS Build: 22621.3085

### Repro Steps:

1. Hit the URL: http://localhost:15888/
2. TAB to "Resources" tab item and hit select the tab by pressing Enter key.
3. Tab through the page and check if 'Filter' edit box is accessible by keyboard.

### Actual Results:
List options under 'Type filter' button are not accessible by keyboard.

### Expected Results:
List options under 'Type filter' button should be accessible using keyboard TAB/ arrow keys.

### User Impact:
Keyboard users will not be able to interact with the control and will miss the information regarding it.

### Attachment:

https://github.com/dotnet/aspire/assets/90901519/00f7c525-4684-4207-9170-e5614ae70ee1

**Comment by adamint** [comment]
@vnbaaij this is an issue with Popover

**Comment by adamint** [comment]
Waiting on 4.6.1 release of fluentui-blazor: https://github.com/microsoft/fluentui-blazor/pull/1800

**Comment by adamint** [comment]
@PoojaNamde could you please re-test? Thank you.

**Comment by adamint** [comment]
@PoojaNamde

**Comment by SwatiJo** [comment]
@adamint It should be navigable using up and down arrow keys

**Comment by adamint** [comment]
Yes @SwatiJo could you re-test and close this bug? We updated a package that should have fixed this issue.

**Comment by SwatiJo** [comment]
@adamint Can you please share link to the package where this issue is fixed?

**Comment by adamint** [comment]
It is fluentui-blazor, but you can re-test with the main branch here. @SwatiJo

**Comment by SwatiJo** [comment]
@adamint Retested this issue on main branch. Options under dropdown are navigable with left arrow key but that is not correct navigation, when dropdown opens, focus should move to dropdown and then options inside dropdown should be navigable using up and down arrow keys.

**Comment by kvenkatrajan** [comment]
I have removed the "needs-author-action" from the a11y bugs since this is moved to stale and closed automatically with no author action.

**Comment by adamint** [comment]
@kvenkatrajan this isn't a resolved issue

**Comment by adamint** [comment]
@dvoituron @vnbaaij do you know why your popover navigation PR isn't working for us?

**Comment by vnbaaij** [comment]
As @tlmii already mentioned in the merged PR:

> The keyboard keys needed are a little awkward - once you open the popover by hitting space you need to hit the left arrow to move back through the checkboxes. But it does work.

This is how it works in the web component. We can't change that on the Blazor side.

**Comment by adamint** [comment]
We were just told by @SwatiJo that this still constitutes a sev2 accessibility issue as the new behavior is not discoverable or correct. @SwatiJo how do we route this to the web components team?

@vnbaaij

**Comment by vnbaaij** [comment]
> We were just told by @SwatiJo that this still constitutes a sev2 accessibility issue as the new behavior is not discoverable or correct. @SwatiJo how do we route this to the web components team?
> 
> @vnbaaij

An issue needs to be created in the microsoft/fluetui repo. It needs to be made clear that this targest the **web-components-v2 ** branch (https://github.com/microsoft/fluentui/tree/web-components-v2/packages/web-components).

**Comment by adamint** [comment]
> > We were just told by @SwatiJo that this still constitutes a sev2 accessibility issue as the new behavior is not discoverable or correct. @SwatiJo how do we route this to the web components team?
> > @vnbaaij
> 
> An issue needs to be created in the microsoft/fluetui repo. It needs to be made clear that this targest the **web-components-v2** branch (https://github.com/microsoft/fluentui/tree/web-components-v2/packages/web-components).

@PoojaNamde please follow these instructions, noting that this affects the Popover comment.

**Comment by dvoituron** [comment]
@vnbaaij @adamint @SwatiJo 
Sorry, I was off yesterday. I think that we worked on this "feature" a few weeks ago, to find [a workaround in the FluentUI-Blazor lib (PR 1800)](https://github.com/microsoft/fluentui-blazor/pull/1800).

After checking, your problem seems to be due to using this `FluentPopover` in a `FluentToolbar` (which seems to take precedence over the JS parts added in this PR 1800). A small workaround for your case could be to replace the "toolbar" with a "stack" and add a style to have the same visual.

```html
<style>
    .toolbar {
        padding: var(--layout-toolbar-padding);
        grid-area: toolbar;
    }
</style>

<FluentStack role="toolbar" Class="toolbar"> ... 
```
![image](https://github.com/dotnet/aspire/assets/8350694/2cd53ece-52f6-41a2-98b2-9cea9b24e39c)

Example using only keys to navigate.
![peek](https://github.com/dotnet/aspire/assets/8350694/fdfc2690-a563-48dc-add1-9d3cbca988ea)

**Comment by dvoituron** [comment]
After some further research, it seems that the web component `fluent-toolbar` changes all sub-elements with `tabindex="-1"`, except the one it's on. And that it overloads the passage from one element to another via the arrows.

So, moving the `FluentPopover` should resolve that. See the PR #4243.

**Comment by SwatiJo** [comment]
@adamint @kvenkatrajan This issue has been fixed. @PoojaNamde  please close this bug
[NoRepro_TypeFilterButton.webm](https://github.com/dotnet/aspire/assets/167469595/722bbb3c-3c63-4cc6-a667-e34ac4c1b301)

**Comment by PoojaNamde** [comment]
#Closed; Issue is fixed as per above comment.
